packages/nx/src/tasks-runner/is-tui-enabled.ts

```diff
@@ -1,59 +1,61 @@
 import type { NxJsonConfiguration } from '../config/nx-json';
-import { readNxJsonFromDisk } from '../devkit-internals';
+import { NxArgs } from '../utils/command-line-utils';
 import { isCI } from '../utils/is-ci';
 
 let tuiEnabled = undefined;
 
-export function isTuiEnabled(
+/**
+ * **MOST** places this function is called will short circuit, and thus not need
+ * to pass nxArgs or nxJson, as the tuiEnabled cache or env var will already be populated from
+ * the call inside `run-command.ts`.
+ *
+ * @returns If tui is enabled
+ */
+export function isTuiEnabled() {
+  return process.env.NX_TUI === 'true';
+}
+
+export function shouldUseTui(
   nxJson?: NxJsonConfiguration,
-  skipCapableCheck = false
+  nxArgs?: NxArgs,
```

Comment:
Can these variables be made required now? Might also be worth mentioning that `isTuiEnabled()` should be used in most places.